pkg/platform/abstract/platform.go

```diff
@@ -132,6 +132,9 @@ func (ap *Platform) HandleDeployFunction(existingFunctionConfig *functionconfig.
 			// use the function configuration augmented by the builder
 			createFunctionOptions.FunctionConfig.Spec.Image = buildResult.Image
 
+			// Update function config with ImagePull secrets so image we just build can be pulled from docker registry
+			createFunctionOptions.FunctionConfig.Spec.ImagePullSecrets = buildResult.UpdatedFunctionConfig.Spec.ImagePullSecrets
```

Comment:
condition if not exists (so you don't override) and move to not depend on `functionBuildRequired`
Also it's pushpull secret (pull for controller, push for kaniko/builder)
And since it's for the kaniko build - need to be enriched before line 121